### DISCUSSION TITLE
请问通义千问有prompt构造的教程吗

### DISCUSSION THREAD

**Discussion by I-am-little-curly-hair** [QwenLM/Qwen#112] [General]
我试了几种常用的prompt，发现无论怎么给定上下文，给定多少上下文，模型都不会参考上下文的信息回答问题，还是我行我素的根据模型能力来生成，请问这种问题怎么解决

**Comment by clawbolt** [comment]
举例说明，【我喜欢看完美的东西】。
文本生成token，存在以下3种可能：
1. 我 喜欢 看完 美的(mei de) 东西
2. 我 喜欢 看完 美的(mei di)  东西
3. 我 喜欢 看 完美的 东西

如果生成的token是第1种情况，就和原句子的真实意思差很多。我还在测试不同的token对模型推理的影响。

**Comment by FaliKurve** [comment]
举例说明

**Comment by MmringZee** [comment]
有同样的感受, 加了很多限定或者需要它在某种default情况下输出特定内容, 不仅无法理解并且输出, 甚至还会打破输出格式

**Comment by youssefofficial159-coder** [comment]
Dear Qwen Team,

I am an Arabic-speaking user who has been actively using Qwen Studio and following the development of the Qwen model family. I have been genuinely impressed by Qwen’s reasoning capabilities, multilingual performance, and particularly its ability to handle complex discussions in Arabic.

I would like to share some detailed suggestions that I believe could make Qwen Studio significantly more powerful for students, researchers, developers, and advanced users.

### 1. Advanced Deep Research Mode

Please consider developing a more autonomous research mode capable of:

* Searching across academic papers, books, official documents, databases, and high-quality web sources.
* Prioritizing primary and authoritative sources rather than relying mainly on secondary websites.
* Comparing conflicting sources and explaining why they disagree.
* Providing precise, claim-level citations.
* Clearly distinguishing established facts, interpretations, hypotheses, and model-generated inferences.
* Performing a final verification pass before presenting the answer.

A particularly valuable feature would be a **"Primary Sources Only"** research option.

### 2. Stronger Source and Citation Verification

Qwen could become much more reliable for serious research by introducing an integrated citation-verification system.

For each important claim, the system could show:

**Claim → Source → Exact supporting passage → Source type → Reliability/context**

It would also be useful if Qwen could automatically detect when a citation does not actually support the claim being made.

### 3. Advanced Arabic Support

Arabic should be treated as a first-class language rather than simply another translation target.

Potential improvements include:

* Better RTL rendering, especially in Arabic-English mixed text.
* Better handling of Arabic technical terminology.
* Stronger understanding of Modern Standard Arabic and classical Arabic.
* Better Arabic voice and dialect support.
* Better Arabic search and retrieval.
* Improved performance on long Arabic documents.
* More accurate generation of structured Arabic tables, equations, and technical material.

An **Arabic-focused research mode** would be particularly valuable.

### 4. Specialized Knowledge and Research Modes

I would strongly encourage Qwen to expand specialized models or modes for domains such as:

* Mathematics
* Physics
* Biology
* Medicine
* Law
* Programming
* Academic research
* History
* Philosophy
* Arabic language and literature
* Religious and classical texts

For specialized domains, the model should use domain-appropriate sources, terminology, reasoning standards, and verification procedures rather than applying exactly the same behavior to every subject.

### 5. Rigorous Islamic and Classical-Text Research

As an Arabic-speaking user, I believe there is significant potential for Qwen to become an excellent research assistant for Islamic and classical Arabic scholarship.

I am **not suggesting that the model should adopt a particular religious position**. Rather, it should provide a rigorous research framework that can represent different scholarly positions accurately.

Useful features could include:

* Search across major classical and contemporary Arabic books.
* Exact book, volume, page, and edition references where available.
* Hadith source identification and grading with attribution to recognized scholars.
* Comparison between scholarly opinions and schools of thought.
* Clear separation between Qur'anic text, hadith, scholarly interpretation, and the model's own reasoning.
* Ability to trace a quotation back to its earliest identifiable source.
* Warning the user when a quotation or attribution cannot be verified.

This would make Qwen much more useful for serious Arabic-language research.

### 6. Personal Research Workspace

A dedicated research workspace could allow users to create projects containing:

* Uploaded books and papers
* Web sources
* Notes
* Previous analyses
* Research questions
* Citations
* Generated summaries
* Structured knowledge graphs

The system could maintain project-level context without requiring the user to repeatedly explain the same background.

### 7. Model Comparison and Automatic Model Routing

As Qwen continues expanding its model family, I would like to see a system that automatically determines which model is best for a task.

For example:

**Reasoning task → reasoning model**
**Coding → coding-optimized model**
**Long document → long-context model**
**Fast/simple question → lightweight model**
**Multimodal task → vision-capable model**

An advanced interface could also allow users to send the same prompt to multiple Qwen models and compare their answers side by side.

### 8. Stronger Multimodal Capabilities

Qwen Studio could become a much more complete multimodal workspace by improving:

* Long video understanding
* Audio understanding and transcription
* Document and PDF analysis
* Scientific diagrams
* Mathematical notation
* Presentations
* Image reasoning
* Video generation

Long-form video generation and editing would be particularly valuable if users could eventually create coherent videos of approximately 10 minutes or longer.

### 9. AI Agents and Task Execution

I would like Qwen Studio to develop more capable agents that can perform multi-step tasks rather than only answer questions.

For example:

**Research → search → collect sources → analyze → verify → write report → create presentation**

Or:

**Study → analyze curriculum → identify weaknesses → create study plan → generate exercises → evaluate answers → adapt the next lesson**

The ability to supervise the agent while retaining user control would be important.

### 10. Advanced Learning Mode

For students, Qwen could offer a dedicated adaptive learning system that understands a curriculum rather than simply answering isolated questions.

It could:

* Follow national curricula.
* Diagnose misconceptions.
* Track mastery of individual concepts.
* Generate progressively harder exercises.
* Use Socratic questioning.
* Analyze mistakes.
* Adapt explanations to the student's level.
* Create revision schedules.
* Simulate examinations.

Support for national education systems, including Arabic-speaking countries, could make Qwen significantly more useful to students.

### 11. Better Long-Term Personalization

An advanced but user-controlled memory system could allow Qwen to remember:

* Learning objectives
* Preferred explanation style
* Long-term projects
* Research interests
* Frequently used terminology
* Previous decisions and constraints

This should always remain transparent and controllable by the user, with clear separation between temporary conversation context and persistent memory.

### 12. A "Verify Before Answering" Option

For questions where accuracy is more important than speed, I would strongly value a mode in which Qwen performs an internal verification workflow before responding:

**Understand → reason → search → cross-check → identify uncertainty → answer**

The model should explicitly state when evidence is insufficient instead of filling gaps with plausible-sounding information.

---

### Final Suggestion

More broadly, I believe Qwen has an opportunity to differentiate itself not only by making models larger or benchmarks higher, but by building a complete **AI research and reasoning ecosystem**.

The combination of:

**strong reasoning + reliable research + primary-source retrieval + specialized expertise + excellent Arabic support + multimodal capabilities + autonomous agents**

could make Qwen Studio one of the most powerful general-purpose AI platforms available to users worldwide.

I appreciate the work your team is doing, and I hope this feedback is useful for future Qwen Studio and model development.

Best regards,
An Arabic-speaking Qwen user

**Reply by youssefofficial159-coder** [reply]
جيد
